src/deterministic_vi/elbo_objective.jl

```diff
@@ -20,18 +20,17 @@ Returns:
 function calculate_source_pixel_brightness!{NumType <: Number}(
                     elbo_vars::ElboIntermediateVariables{NumType},
                     ea::ElboArgs{NumType},
-                    E_G_s::SensitiveFloat{CanonicalParams, NumType},
-                    var_G_s::SensitiveFloat{CanonicalParams, NumType},
-                    fs0m::SensitiveFloat{StarPosParams, NumType},
-                    fs1m::SensitiveFloat{GalaxyPosParams, NumType},
+                    E_G_s::SensitiveFloat{NumType},
+                    var_G_s::SensitiveFloat{NumType},
+                    fs0m::SensitiveFloat{NumType},
```

Comment:
I see your point. At first I thought those kind of checks were a good use of the type system. But when I realized that that was causing Julia to compile a different version for each `ParamSet` it got called with, I wasn't so sure.  I do like now how `SensitiveFloat` isn't specific to any other Celeste data types. I'll just add some length asserts for the time being.